### PR TITLE
feat: factorise download logic

### PR DESCRIPTION
**Pull request by axel7083**
## Description

Follow-up works to split https://github.com/containers/podman-desktop-extension-minikube/pull/156

## Notable change

As explained in https://github.com/containers/podman-desktop-extension-minikube/pull/156 we have two way of downloading the binary for the minikube extension.

The old way `minikube-installer.ts` and the new way `download.ts`. (currently both implemented and used: this pr clean up that)

The existing `CliTool` uses the new way, which is more robust and more tested. This PR remove the old way, to use the new way.

- Deleting `minikube-installer.ts` and associate code
- Making the install command uses the existing `download.ts` instead 

## How to tests

- [x] Lot of unit tests added 

**Manually**

1. uninstall any minikube on the system
2. install through the status bar item
3. assert cli tool page updated

See video to reproduce

https://github.com/user-attachments/assets/c60d98a6-3529-4e15-b787-7e1afdce930e

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -248,51 +247,88 @@ async function createProvider(
...
+        progress.report({ message: 'Installing minikube system-wide' });
+        const systemPath = await installBinaryToSystem(destFile, 'minikube');
+        minikubeCliTool?.updateVersion({
+          version: lastReleaseVersion,
+          path: systemPath,
+        });
+        minikubeCliToolUpdaterDisposable?.dispose();
+      },
+    );
+  });
```
[the remark this answers is not in the document]
Comment (axel7083) [reply src/extension.ts:297]: Oh yeah you are right, thanks I will make the change

[the remark this answers is not in the document]
Comment (axel7083) [reply src/extension.ts]: another problem could have occur, what if the user cancel the system wide install, we still need to update the cli tool.

So the solution is the following 

https://github.com/containers/podman-desktop-extension-minikube/blob/4c45f5872c16ab32109ae0150d1444032e11e636/src/extension.ts#L280-L293

(already commited)

We try catch finally, so we ensure even tho the user cancel the system wide install, we update the cli tool with the extension-folder path

**Comment by benoitf** [comment]
in the description it is saying that it's refactoring the download but in the PR I see detectMinikube is being renamed to findMinikube and the content is also changed so I think the scope is bigger than what the description is mentioning

Also as it's involving lot of different changes the review is harder then no one has reviewed it yet

File: src/util.ts [hunk 2]
```diff
@@ -52,36 +52,17 @@ export function getMinikubeHome(): string | undefined {
...
-        PATH: getMinikubePath(),
-        MINIKUBE_HOME: getMinikubeHome(),
-      },
-    });
-    return 'minikube';
-  } catch (e) {
-    // ignore and try another way
+    return await whereBinary('minikube');
+  } catch (err: unknown) {
+    console.debug(err);
```
Comment (jeffmaury) [inline src/util.ts:59]: To be removed ?

**Comment by axel7083** [comment]
Sorry this PR is outdated, and should be closed

**Pull request closed without merging**